### PR TITLE
Issue 5161: Added dmSocket::SetQuickAck

### LINKED ISSUE #5161: Add support for TCP_QUICKACK in dmSocket
As an extra optimization for websockets, we want to disable the ACK.
For unsupported platforms, it will be a no-op

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/dlib/src/dlib/socket.cpp [hunk 1]
```diff
@@ -821,6 +821,15 @@ namespace dmSocket
         return SetSockoptBool(socket, IPPROTO_TCP, TCP_NODELAY, no_delay);
     }
 
+    Result SetQuickAck(Socket socket, bool use_quick_ack)
+    {
+#if defined(__MACH__)
```
Comment (JCash) [inline engine/dlib/src/dlib/socket.cpp:826]: I pushed now, to let CI decide if any other platforms fail using this code.

File: engine/dlib/src/test/test_socket.cpp [hunk 2]
```diff
@@ -135,6 +135,9 @@ inline dmSocket::Socket GetSocket(dmSocket::Domain domain)
     result = dmSocket::SetNoDelay(instance, true);
     if (result != dmSocket::RESULT_OK) return -1;
 
+    result = dmSocket::SetQuickAck(instance, true);
```
Comment (britzl) [inline engine/dlib/src/test/test_socket.cpp:138]: Should this be configurable perhaps and not always on? Are there disadvantages of having quick ACK always enabled? I'm thinking of usecase when the socket is used for an HTTP connection vs a TCP connection with a lot of data going back and forth (games)?

LuaSocket has a setoption() function with options such as "keepalive" and "tcp-nodelay" as valid choices.

**Review by britzl — APPROVED** [review]

File: engine/dlib/src/test/test_socket.cpp (hunk 2 above)
Comment (JCash) [reply engine/dlib/src/test/test_socket.cpp:138]: I only set it here in the tests to make sure it gets used.
In the engine itself we use whatever is the default.

Comment (JCash) [reply engine/dlib/src/test/test_socket.cpp:138]: Or perhaps I'm missing your point :)

Comment (britzl) [reply engine/dlib/src/test/test_socket.cpp:138]: Ah, I'm confused. It was a test :-)

**Pull request merged**